web/src/stream_settings_ui.js

```diff
@@ -67,6 +67,16 @@ export function update_left_panel_row(sub) {
         $new_row.addClass("active");
     }
 
+    // Show spinner in $new_row if $row contains spinner.
+    const show_spinner = $($row).find(".sub_unsub_status").children().length !== 0;
+    if (show_spinner) {
+        stream_settings_components.display_subscribe_toggle_spinner($new_row);
+        // Hide toggle spinner for after 300ms.
+        setTimeout(() => {
+            stream_settings_components.hide_subscribe_toggle_spinner($new_row);
+        }, 300);
+    }
```

Comment:
In case of error, the spinner will keep on loading in this case because update_left_panel_row will not be called in case of error, since update_left_panel_row is called when an appropriate event is received. There will be no event in case of error.

Also, I don't like the idea of having the logic to hide the spinner depend on events when it doesn't need to depend on it.

Let's keep the logic for showing and hiding the spinner where the request is being made. 
You can track whether minimum time has elapsed in a variable, and whether the request has completed in another variable. The spinner should be shown when both of them are true. 

I couldn't find a mechanism in loading.js to have minimum time for the loading spinner, but I might be wrong and it's better to check on CZO since it was implied that this mechanism already exists. If not, we still might want to make it part of loading.js, so that it's easier for future contributors to use that mechanism. That conversation needs to happen for this PR.

Reply:
> Also, I don't like the idea of having the logic to hide the spinner depend on events when it doesn't need to depend on it.

I agree. But if we still require `update_left_panel_row` to wait when the loader is still loading.

In loading.js, we need functions to check in loader is loading, add callback function when loader ends. I've started a discussion in [CZO](https://chat.zulip.org/#narrow/stream/6-frontend/topic/missing.20features.20in.20loading.2Ets/near/1908697) regarding loader.ts